lib/jarvis/workpool.rb

```diff
@@ -0,0 +1,52 @@
+require "concurrent"
+
+class WorkPool
+  ADMINISTRATIVE = :ADMINISTRATIVE
+  NORMAL = :NORMAL
+
+  class InvalidWorkPoolName < StandardError; end
+
+  class << self
+    def setup_singleton
+      @singleton = self.new
+      nil
+    end
+
+    def fetch(name)
+      @singleton.fetch(name)
+    end
+
+    def post(name, &block)
+      fetch(name).post(&block)
+    end
+  end
```

Comment:
Any specific reason not to use the `Singleton` module? This will make sure that new can't be called.